### PR TITLE
[NF] Add DiSCo challenge data fetcher

### PR DESCRIPTION
**Pull request by skoudoro**
This PR adds DiSCo challenge data fetcher for tractography tutorials, benchmarks and tests.

More information about this dataset in the following link: DOI: 10.17632/fgf86jdfg6.2

4 fetchers has been added:

- `fetch_disco_dataset`  -> 4.94 GB (contains disco1 + disco2+disco3)
- `fetch_disco1_dataset` -> 1.65 GB
- `fetch_disco2_dataset` -> 1.65 GB
-  `fetch_disco3_dataset` -> 1.65 GB

### CODE REVIEW

**Comment by gabknight** [comment]
Thanks @skoudoro for preparing this. 

I noticed typos in 6 file names. I corrected the filenames in the repo, but it needs to be approved by Mendeley data (up to 2 business days). The new URL will be https://data.mendeley.com/datasets/fgf86jdfg6/3

When I fetch the data with the fetcher I got a timeout error: [timeout_error.txt](https://github.com/dipy/dipy/files/14938082/timeout_error.txt). The error appeared twice for disco1, 0 for disco2, and 8 times for disco3, until I had all data. `dipy.data.fetcher.fetch_disco_dataset()` worked fine. I don't know why did is happening.

File: dipy/data/fetcher.py [hunk 1]
```diff
@@ -721,6 +780,336 @@ def fetcher():
...
+     'lowRes_DiSCo1_Strand_Average_Diameter.nii.gz',
+     'lowRes_DiSCo1_DWI_RicianNoise-snr10.nii.gz',
+     'highRes_DiSCo1_Strand_ODFs.nii.gz',
+     'highRes_DiSCo1_DWI_RicianNoise-snr50.nii.gz',
+     'highRes_DiSCo1_DWI.nii.gz',
+     'highRes_DiSCo1_ROIs.nii.gz',
+     'highRes_DiSCo1_DWI_RicianNoise-snr40.nii.gz',
+     'highRes_DiSCo1_mask.nii.gz',
+     'highRes_DiSCo1_Strand_Bundle_Count.nii.gz',
+     'highRes_DiSCo1_DWI_RicianNoise-snr20.nii.gz',
```
[the remark this answers is not in the document]
Comment (skoudoro) [reply dipy/data/fetcher.py:849]: I plan to do a new PR later to allow optional data.

File: dipy/data/fetcher.py [hunk 2]
```diff
@@ -721,6 +780,336 @@ def fetcher():
...
+     'e8911b580e7fc7e2d94d238f2c81c43b', '647714842c8dea4d192bdd7c2b41fd3b',
+     'bba1e4283ed853eb484497db4b6da460', 'd761384f8fed31c53197ce4916f32296',
+     '3b119c1b8487da71412f3275616003dc', '47c300fcb48b882c124d69086b224956',
+     '48f844071bc1ca4d740d42620bd7d2ad', '9e47185ae517f6f35daa1aa16c51ba45'],
+    doc=('Download DISCO 3 dataset: The Diffusion-Simulated Connectivity '
+         'Dataset. DOI: 10.17632/fgf86jdfg6.2'),
+    data_size='1.65 GB')
+
 
 def get_fnames(name='small_64D'):
```
[the remark this answers is not in the document]
Comment (skoudoro) [reply dipy/data/fetcher.py:1117]: done

**Comment by skoudoro** [comment]
Hi @gabknight, 

I think I have addressed all your comments. When the CI's finish, feel free to look at this again.

**Comment by skoudoro** [comment]
Not sure why all tests failed in this PR.... investigating...

**Comment by gabknight** [comment]
Works perfectly! Thanks for doing this @skoudoro

**Comment by skoudoro** [comment]
Thanks for approving @gabknight. So I am going ahead and merge this PR.

**Pull request merged**